### ISSUE TITLE
Loss remains stable around 2.0 on 12h long training

### ISSUE THREAD

**Issue by devstermarts** [moiseshorta/RAVE-Latent-Diffusion#5]
Hi Moises. Not really an issue, more a question: I've been observing that some (most) of my RAVE LD trainings tend to oscillate around a train and validation loss of 2.0 even though they're running for hours. Is this an expected behaviour?

In a few earlier trainings, the loss went down straight close to 0 but I cannot reconstruct why that was. 
Any ideas or suggestions would be much appreciated 😸 

Best, Martin

**Comment by moiseshorta** [comment]
Hey Martin,

Would you mind posting some more details? Aproximately how long is your dataset? How many dimensions is your RAVE model and which version are you using? 

I need to do further trainings, but generally speaking, on a dataset of about 10+hrs, the loss should definitely go bellow 2.0 and slowly drop. Did you change the LR? Also, I've noticed that if the dataset is quite heterogeneous, you might be getting that sort of behaviour, i.e. some latent codes only contain non-rhythmic music, while others are having some very rhythmic parts.

Best,
M

**Comment by devstermarts** [comment]
Hi Moises, thanks for the fast reply. Certainly, here are the details: 

The model is a RAVE V1 model with 32 dimensions (i've been using the rave_dims==32 for training and generation accordingly). I've been training the model with a higher than default num_signal value (5x) and capacity (1.5x). 
The training set is roughly 1.5-2h. Going back to the logs, it seems, however, that only two 1MB .npy files have been created in the latent folder, considering the file names, it seems like the majority from the files have been skipped for some reason. I missed that on my earlier investigation. I've used a latent_length of 8192 in dataset preparation. 
The training has been per default configuration, i've only changed save_interval to 200. 

Does that help narrow it down for an intuition of what's happening?

Thanks and all best,
Martin

**Comment by moiseshorta** [comment]
Yes, it seems your latent dataset is too small. By default the preprocessing script skips audios which are smaller than the context window size. So considering you're using a context window of 8192, the audios should be at least 349.52 seconds long (or 5.84 minutes) at 48Khz.

Hope this helps.

**Comment by devstermarts** [comment]
That did the trick. Reducing latent_length to 2048 -> 13 epochs in and we're at a loss of 1.4 already. Thank you for the clarification.
